obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+                        int(head_line[19:20]) * 100000) + add_seconds
+            return True
+        except:
+            return False
+    else:
+        return False
+
+
+def _int_conv(string):
+    """
+    Convenience tool to convert from string to integer.
+    If empty string return a 999 rather than an error.
+
+    >>> _int_conv('12')
+    12
+    >>> _int_conv('')
+    999
+    """
+    try:
+        intstring = int(string)
+    except:
+        intstring = 999
+    return intstring
+
+
+def _float_conv(string):
+    """
+    Convenience tool to convert from string to float.
+    If empty string return NaN rather than an error.
+
+    >>> _float_conv('12')
+    12.0
+    >>> _float_conv('')
+    999.0
+    >>> _float_conv('12.324')
+    12.324
+    """
+    try:
+        floatstring = float(string)
+    except:
+        # cjh Changed this to deal with QuakeML validation issues with 'NaN'
+        floatstring = float('999')
+    return floatstring
+
+
+def _str_conv(number, rounded=False):
+    """
+    Convenience tool to convert a number, either float or int into a string.
+    If the int is 999, or the float is NaN, returns empty string.
+
+    >>> _str_conv(12.3)
+    '12.3'
+    >>> _str_conv(12.34546, rounded=1)
+    '12.3'
+    """
+    string = str(number)
+    if (isinstance(number, float) and np.isnan(number)) or number == 999:
+        string = ' '
+    elif not rounded and \
+            (isinstance(number, float) or isinstance(number, int)):
+        if number < 100000:
+            string = str(number)
+        else:
+            exponant = int('{0:.2E}'.format(number).split('E+')[-1]) - 1
+            divisor = 10 ** exponant
+            string = '{0:.1f}'.format(number / divisor) + 'e' + str(exponant)
+    elif rounded == 2 and \
+            (isinstance(number, float) or isinstance(number, int)):
+        string = '{0:.2f}'.format(number)
+    elif rounded == 1 and \
+            (isinstance(number, float) or isinstance(number, int)):
+        string = '{0:.1f}'.format(number)
+    else:
+        return str(number)
+    return str(string)
+
+
+def _evmagtonor(mag_type):
+    """
+    Switch from obspy event magnitude types to seisan syntax.
+
+    >>> _evmagtonor('mB')
+    'b'
+    >>> _evmagtonor('M')
+    'W'
+    >>> _evmagtonor('bob')
+    ''
+    """
+    mag_type = str(mag_type)
+    if mag_type in ['ML', 'MLv']:
+        # MLv is local magnitude on vertical component
+        mag = 'L'
+    elif mag_type == 'mB':
+        mag = 'b'
+    elif mag_type == 'Ms':
+        mag = 's'
+    elif mag_type == 'MS':
+        mag = 'S'
+    elif mag_type in ['MW', 'Mw']:
+        mag = 'W'
+    elif mag_type == 'MbLg':
+        mag = 'G'
+    elif mag_type in ['Mc', 'MC']:
+        mag = 'C'
+    elif mag_type == 'M':
+        mag = 'W'  # Convert generic magnitude to moment magnitude
+        msg = ('Converting generic magnitude to being stored as moment mag')
+        warnings.warn(msg)
```

Comment:
This could be simplified with 

```python
mapping = {
    "ML": "L",
    "MS": "S", ...}

mag = mapping[mag_type] if mag_type in mapping else ''
```